clinica/iotools/converters/oasis3_to_bids/oasis3_to_bids.py

```diff
@@ -57,16 +57,35 @@ def convert_clinical_data(self, clinical_data_dir, bids_dir):
                 else:
                     sessions_dict[y][z]["diagnosis"] = "CN"
 
+        for bids_id in bids_ids:
+            bids_subj_folder = path.join(bids_dir, bids_id)
+            session_paths = glob(path.join(bids_subj_folder, "*"))
+            for s in session_paths:
+                # Get the session name in month differences
+                s_name = path.basename(s).replace("ses-", "")
+                diff_months = str(sessions_dict[bids_id][s_name]['diff_months'])
+                s_new_name = "M" + diff_months
+                # Replace the name of the sessions in the dictionary
+                sessions_dict[bids_id][s_new_name] = sessions_dict[bids_id].pop(s_name)
+                sessions_dict[bids_id][s_new_name]['session_id'] = "ses-" + s_new_name
+                # Replace it in for the folder
+                s_new_path = s.replace(s_name, s_new_name)
+                os.rename(s, s_new_path)
+                # Replace it for each file in its folder
+                file_paths = glob(path.join(s_new_path, "**/*"))
+                for f in file_paths:
+                    os.rename(f, f.replace(s_name, "M" + diff_months))
```

Comment:
Since the format of `diff_months` is numeric, we probably need to apply this formating rule:

```suggestion
                    os.rename(f, f.replace(s_name, f"M{diff_months:02d}"))
```

to translate `0` to `M00` or `6` to `M06`.

Reply:
I may be wrong though so feel free to reject the suggestion if my assumption is incorrect.

Reply:
Ah top ! Je cherchais un truc comme ça ! Par contre, mes tests n'ont pas l'air de fonctionner:

![image](https://user-images.githubusercontent.com/27341903/125816794-97175b4d-52d1-4715-a02d-3e18efb23bf3.png)
![image](https://user-images.githubusercontent.com/27341903/125816853-0dae093c-7825-4428-bd50-58e5d7fda9e7.png)

Reply:
Looks like `diff_months` is in string format instead of being numerical. Try casting it to `int` by replacing `diff_months` with `int(diff_months)` in my suggestion.

Reply:
Forget adding the `int` cast and look at my suggestion for not casting `diff_months` to string earlier in the code.

Reply:
Then you should be able to reuse the new session name here:

```suggestion
                    os.rename(f, f.replace(s_name, s_new_name))
```